src/rulesets/LinearAlgebra/symmetric.jl

```diff
@@ -279,6 +279,172 @@ function rrule(::typeof(svdvals), A::LinearAlgebra.RealHermSymComplexHerm{<:BLAS
+        end
+
+        function rrule(::typeof($func), A::LinearAlgebra.RealHermSymComplexHerm)
+            Y, cache = _matfun($func, A)
+            $(Symbol("$(func)_pullback"))(ΔY::AbstractZero) = (NO_FIELDS, ΔY)
+            function $(Symbol("$(func)_pullback"))(ΔY)
+                # for Hermitian Y, we don't need to realify the diagonal of ΔY, since the
+                # effect is the same as applying _hermitrize! at the end
+                ∂Y = eltype(Y) <: Real ? real(ΔY) : ΔY
+                # for matrix functions, the pullback is related to the pushforward by an adjoint
+                Ā = _matfun_frechet($func, A, Y, ∂Y', cache)
+                # the cotangent of Hermitian A should be Hermitian
+                ∂A = typeof(A)(eltype(A) <: Real ? real(Ā) : Ā, A.uplo)
+                _hermitrize!(∂A.data)
+                return NO_FIELDS, ∂A
+            end
+            return Y, $(Symbol("$(func)_pullback"))
+        end
+    end
+end
+
+function frule((_, ΔA), ::typeof(sincos), A::LinearAlgebra.RealHermSymComplexHerm)
+    ΔA isa AbstractZero && return sincos(A), ΔA
+    sinA, (λ, U, sinλ, cosλ) = _matfun(sin, A)
+    cosA = _symhermtype(sinA)((U * Diagonal(cosλ)) * U')
+    tmp = ΔA * U
+    ∂Λ = U' * tmp
+    ∂sinΛ = _muldiffquotmat!(similar(∂Λ), sin, λ, sinλ, cosλ, ∂Λ)
+    ∂cosΛ = _muldiffquotmat!(∂Λ, cos, λ, cosλ, -sinλ, ∂Λ)
+    ∂sinA = _symhermlike!(mul!(∂sinΛ, U, mul!(tmp, ∂sinΛ, U')), sinA)
+    ∂cosA = _symhermlike!(mul!(∂cosΛ, U, mul!(tmp, ∂cosΛ, U')), cosA)
+    Y = (sinA, cosA)
+    ∂Y = Composite{typeof(Y)}(∂sinA, ∂cosA)
+    return Y, ∂Y
+end
+
+function rrule(::typeof(sincos), A::LinearAlgebra.RealHermSymComplexHerm)
+    sinA, (λ, U, sinλ, cosλ) = _matfun(sin, A)
+    cosA = typeof(sinA)((U * Diagonal(cosλ)) * U', sinA.uplo)
+    Y = (sinA, cosA)
+    sincos_pullback(ΔY::AbstractZero) = (NO_FIELDS, ΔY)
+    function sincos_pullback(ΔY::Composite)
+        ΔsinA, ΔcosA = ΔY
+        ΔsinA isa AbstractZero && ΔcosA isa AbstractZero && return NO_FIELDS, ΔsinA + ΔcosA
+        if eltype(A) <: Real
+            ∂sinA, ∂cosA = real(ΔsinA), real(ΔcosA)
+        else
+            ∂sinA, ∂cosA = ΔsinA, ΔcosA
+        end
+        if ∂cosA isa Zero
+            Ā = _matfun_frechet(sin, A, sinA, ∂sinA, (λ, U, sinλ, cosλ))
+        elseif ∂sinA isa Zero
+            Ā = _matfun_frechet(cos, A, cosA, ∂cosA, (λ, U, cosλ, -sinλ))
+        else
+            tmp = ∂sinA * U
+            ∂sinΛ = U' * tmp
+            ∂cosΛ = U' * mul!(tmp, ∂cosA, U)
+            ∂Λ = _muldiffquotmat!(∂sinΛ, sin, λ, sinλ, cosλ, ∂sinΛ)
+            ∂Λ = _muldiffquotmat!(∂Λ, cos, λ, cosλ, -sinλ, ∂cosΛ, true)
+            Ā = mul!(∂Λ, U, mul!(tmp, ∂Λ, U'))
+        end
+        _hermitrize!(Ā)
+        ∂A = typeof(A)(Ā, A.uplo)
+        return NO_FIELDS, ∂A
+    end
+    return Y, sincos_pullback
+end
+
+# compute the matrix function f(A), returning also a cache of intermediates for computing
+# the pushforward or pullback.
```

Comment:
```suggestion
# the pushforward or pullback.
# Note any function `f` used with this **must** have a `frule` defined on it.
```

What do you thing about making this a docstring?

Reply:
Makes sense. I intend to overload these functions to support other matrix types in LinearAlgebra for future PRs.